x/leverage/types/weighted_coin.go

```diff
@@ -72,8 +70,8 @@ func (wnp WeightedNormalPair) before(b WeightedNormalPair) bool {
 		return wnp.Borrow.Weight.GT(b.Borrow.Weight)
 	}
 	if wnp.Collateral.Asset.Denom != b.Collateral.Asset.Denom {
-		// if both weight are equal, sort by collateral denom, alphabetical
-		return wnp.Collateral.Asset.Denom != b.Collateral.Asset.Denom
+		// if both weights are equal, sort by collateral denom, alphabetical
+		return wnp.Collateral.Asset.Denom < b.Collateral.Asset.Denom
```

Comment:
This could have created inconsistent ordering of assets - although only ones with identical weights, so it wouldn't have affected positions. Still a bug though.